src/main.c

```diff
@@ -22,8 +22,15 @@
 #include "apdu.h"
 #include "menu.h"
 
+// Swap feature
+#include "swap_lib_calls.h"
+#include "handle_swap_sign_transaction.h"
+#include "handle_get_printable_amount.h"
+#include "handle_check_address.h"
+
 ApduCommand G_command;
 unsigned char G_io_seproxyhal_spi_buffer[IO_SEPROXYHAL_BUFFER_SIZE_B];
+volatile bool G_called_from_swap;
 
 static void reset_main_globals(void) {
     MEMCLEAR(G_command);
```

Comment:
not specific to your PR, but I think MEMCLEAR(G_command); should be added before the line 51 in this file, otherwise host can send a bogus packet, exception will be thrown, then if host continues with P_EXTEND Ledger won't detected that original packet had an error

Reply:
Added